### PR TITLE
Update MD043 to fix required header checks in middle and last when using special char "*"

### LINKED ISSUE #394: MD043 - Using the special value `"*"` results in subsequent required header values being ignored
When using a special value `"*"` for zero or more headings, subsequent required headings in the rule are ignored.  

For example, if you change the following line to any value, the tests will continue to pass: https://github.com/DavidAnson/markdownlint/blob/main/test/required-headings-zero-or-more.json#L9 .  

I.e.

```json
{
  "default": true,
  "MD043": {
    "headings": [
      "# One",
      "*",
      "### ANY VALUE HERE1",
      "*",
      "### ANY VALUE HERE2",
      "*",
      "#### ANY VALUE HERE3"
    ]
  }
}
```

This markdown will still show no errors regardless of whether the "ANY VALUE HERE" header is present or not.
```
# One

## Two

### THREE

## four

## Five

### SiX

#### 7
```

This looks to be related to the change here: https://github.com/DavidAnson/markdownlint/commit/e20502c494c8b8d7f2640c141343c176af4363b6 .

### PR DESCRIPTION
**Pull request by jeremysuriel**
Fixes #394

### CODE REVIEW

**Review by DavidAnson — CHANGES_REQUESTED** [review]
This is awesome, thanks! Love the new test cases. :) I left a few pretty minor comments. FYI, the CI failure is because you didn't update the browser build - to do so, `npm run ci` and add the changed file to the PR.

File: test/required-headings-missing-last-zero-or-more.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+### THREE
+
+## four
+
+## Five
+
+### SiX
+
+#### FOO {MD043:14}
```
[the remark this answers is not in the document]
Comment (jeremysuriel) [reply test/required-headings-missing-last-zero-or-more.md:13]: Will do

File: lib/md043.js [hunk 2]
```diff
@@ -28,8 +28,11 @@ module.exports = {
           const actual = levels[heading.tag] + " " + content;
           const expected = getExpected();
           if (expected === "*") {
-            matchAny = true;
-            getExpected();
+            const nextExpected = getExpected();
+            if (nextExpected.toLowerCase() !== actual.toLowerCase()) {
```
[the remark this answers is not in the document]
Comment (jeremysuriel) [reply lib/md043.js:32]: Given we log actual below and likely want to keep it in the original case there and that only a single execution would occur here or below, I would personally keep it the way it is for better readability.  But I can make this change if you prefer.

[the remark this answers is not in the document]
Comment (DavidAnson) [reply lib/md043.js:32]: Good point, leave it how it was, then.

**Pull request merged**